mobx_codegen/lib/src/template/observable.dart

```diff
@@ -6,13 +6,24 @@ class ObservableTemplate {
   String type;
   String name;
   bool isPrivate;
+  bool isReadOnly;
+
+  /// Formats the `name` from `_foo_bar` to `foo_bar`
+  /// such that the getter gets public
+  String get _getterName {
+    if (isReadOnly) {
+      final nameWithoutUnderline = name.replaceAll(RegExp(r'^_*'), '');
+      return nameWithoutUnderline;
+    }
+    return name;
+  }
 
   @override
   String toString() => """
   final $atomName = Atom(name: '${storeTemplate.parentTypeName}.$name');
 
   @override
-  $type get $name {
+  $type get $_getterName {
```

Comment:
If its readonly, you need to throw an error in the setter. Don't see that getting handled here

Reply:
Actualy it's readonly from the perspective of the client of the store, but it must be modifiable by the store. The setter must stay as is, otherwise it wouldn't be possible to change the value from the variable. The way I did it was to keep the setter private.

Reply:
Because the variable must be declared as private (e.g.: `@readonly int _counter`) there isn't any setter publicly available.

Reply:
So there is still the responsibility of declaring the observable as private at the Store level. That's a hidden assumption and needs to be checked and verified in the codegen too. I am still debating if this is necessary or can be managed in user code itself.

Reply:
That's true. It would be nice to see other people's opinion about this. For now I can't think of a way to make it such that declaring as private isn't required.

Reply:
At least if you declared as public should still work. Maybe we can throw a error during generation to inform the error? The same way as actions are not allowed to be used on setters.

Reply:
Yes we can do those checks but also ensure the user is not adding their own public setter

Reply:
Didn't thought about that, good idea. However I'm not really used to codegen, so may take a little while xd.

Reply:
I managed to disallow public variables, but how can I avoid the setters? Should I search for setters with similar name to the variable?

Reply:
yes there should be a way to check the setter on the same name

Reply:
Done. My solution is basicly caching the public setters found and then checking if there's any that matches the name of a readonly observable. Not sure if that's the right way, but it works.